File: python/GPU.py
from Komponen import Komponen

class GPU(Komponen):
    def __init__(self, vram=0, tipe_gpu="", clock_speed_ghz=0.0, merk="", nama=""):
        super().__init__(merk, nama)
        self.vram = vram
        self.tipe_gpu = tipe_gpu
        self.clock_speed_ghz = clock_speed_ghz
    
    def set_vram(self, vram):
        self.vram = vram
    
    def set_tipe_gpu(self, tipe_gpu):
        self.tipe_gpu = tipe_gpu
    
    def set_clock_speed_ghz(self, clock_speed_ghz):
        self.clock_speed_ghz = clock_speed_ghz
    
    def get_vram(self):
        return self.vram
    
    def get_tipe_gpu(self):
        return self.tipe_gpu
    
    def get_clock_speed_ghz(self):
        return self.clock_speed_ghz
    
    def display_info(self):
        super().display_info()
        print(f" ({self.vram} GB VRAM) {self.tipe_gpu} ~{self.clock_speed_ghz}GHz", end="")

File: cpp/GPU.cpp
#ifndef GPU_CPP
#define GPU_CPP

#include <iostream>
#include <string>
#include "Komponen.cpp"

using namespace std;

class GPU : public Komponen {
private:
    int vram;
    string tipeGPU;
    float clockSpeedGHz;
public:
    GPU() {}

    GPU(int vram, string tipeGPU, float clockSpeedGHz, string merk, string nama) : Komponen(merk, nama) {
        this->vram = vram;
        this->tipeGPU = tipeGPU;
        this->clockSpeedGHz = clockSpeedGHz;
    }

    void setVram(int vram) {
        this->vram = vram;
    }

    void setTipeGPU(string tipeGPU) {
        this->tipeGPU = tipeGPU;
    }

    void setClockSpeedGHz(float clockSpeedGHz) {
        this->clockSpeedGHz = clockSpeedGHz;
    }

    int getVram() {
        return this->vram;
    }

    string getTipeGPU() {
        return this->tipeGPU;
    }

    float getClockSpeedGHz() {
        return this->clockSpeedGHz;
    }

    void displayInfo() override {
        Komponen::displayInfo();
        cout << " (" << vram << " GB VRAM) " << tipeGPU << " ~" << clockSpeedGHz << "GHz";
    }

    ~GPU() {}
};

#endif // GPU_CPP